File: myskills/frontend/src/types/knowledge_base.ts
export interface KnowledgeBase {
  id: string;
  name: string;
  description: string;
  document_count: number;
  vector_count: number;
  summary?: string;
  created_at: string;
  updated_at?: string;
}

export interface CreateKnowledgeBaseData {
  name: string;
  description: string;
}

File: myskills/backend/app/models/knowledge_base.py
from sqlalchemy import Column, String, Integer, Text, DateTime
from sqlalchemy.sql import func
from app.database import Base
import uuid


class KnowledgeBase(Base):
    """知识库模型"""
    __tablename__ = "knowledge_bases"
    
    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(255), nullable=False)
    description = Column(Text)
    document_count = Column(Integer, default=0)
    vector_count = Column(Integer, default=0)
    summary = Column(Text)
    summary_updated_at = Column(DateTime)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
